haystack/dataclasses/streaming_chunk.py

```diff
@@ -70,13 +74,18 @@ class StreamingChunk:
 
     :param content: The content of the message chunk as a string.
     :param meta: A dictionary containing metadata related to the message chunk.
+        NOTE: Both 'finish_reason' field and meta['finish_reason'] are supported for flexibility.
+        The dedicated field provides better type safety and IDE support.
```

Comment:
I'd probably drop this, since we aren't guaranteeing that `finish_reason` will also always be available in the metadata.